camx: package provide all components of the Qualcomm Camera X (CamX)

gkhose-qipl opened the pull request:
The camx-lib package provides all camera libraries, shared across all components of the Qualcomm Camera X (CamX).
This recipe help to create separate camx packages.
This PR support QLI 1.6 features for KLM targets

lumag commented on recipes-multimedia/camx/camx-kt_1.0.1.bb:
```diff
@@ -0,0 +1,21 @@
+SUMMARY = "Qualcomm camera core driver and pipeline related libraries"
```
Please write a proper commit message. Describe your design decisions, issues, etc. It should be a text, rather than a bullet list.

lumag commented on recipes-multimedia/camx/camx-kt_1.0.1.bb:
```diff
@@ -0,0 +1,21 @@
+    install -d ${D}${includedir}
+    cp -r ${S}/usr/include/* ${D}${includedir}
+}
+FILES:${PN} = "/usr/lib/*"
```
Empty line, please

lumag commented on recipes-multimedia/camx/camx-kt_1.0.1.bb:
```diff
@@ -0,0 +1,21 @@
+    cp -r ${S}/usr/include/* ${D}${includedir}
+}
+FILES:${PN} = "/usr/lib/*"
+FILES:${PN}-dev = "/usr/include/*"
```
Last time, I think, we asked for a comment describing the reasons for the override. I see none.

lumag commented on recipes-multimedia/camx/camxfirmware-kt_1.0.1.bb:
```diff
@@ -0,0 +1,21 @@
+    install -d ${D}/lib/firmware
+    cp -r ${S}/lib/firmware/* ${D}/lib/firmware
+}
+FILES:${PN} = "/lib/firmware/*"
```
${nonarch_base_libdir}/firmware

gkhose-qipl commented on recipes-multimedia/camx/camx_1.0.1.bb:
```diff
@@ -0,0 +1,22 @@
+FILES:${PN} = "\
+    /usr/lib/*"
+FILES:${PN}-dev = "/usr/include/*"
+#Skips check for .so symlinks
```
> This is not an answer to the "why" question

can you please review recipes-multimedia/camx/camxfirmware_1.0.1.bb
if description is fine can add in same way for all recipes

lumag commented on recipes-multimedia/camx/camxlib-kt_1.0.1.bb:
```diff
@@ -0,0 +1,25 @@
+SRC_URI = "https://qartifactory-edge.qualcomm.com/artifactory/qsc_releases/software/chip/component/camx.qclinux.0.0/251120/prebuilt_yocto/${BPN}_${PV}_armv8-2a.tar.gz;subdir=${BPN}-${PV}"
+SRC_URI[sha256sum] = "c0b2f23c0c87df6b113466b83948cf2baf80b38492e1899984193108a0bab8e3"
+
+DEPENDS += "camxfirmware fastrpc protobuf-native protobuf protobuf-native libxml2  virtual/egl virtual/libgles2"
```
No, libs don't depend on the firmware. Especially not at the build time.

lumag commented on recipes-multimedia/camx/chicdk-kt_1.0.1.bb:
```diff
@@ -0,0 +1,26 @@
+# no top-level dir in the archive, unpack to subdir to prevent UNPACKDIR pollution
+SRC_URI = "https://qartifactory-edge.qualcomm.com/artifactory/qsc_releases/software/chip/component/camx.qclinux.0.0/${PROMO_TAG}/prebuilt_yocto/${BPN}_${PV}_armv8-2a.tar.gz;subdir=${BPN}-${PV}"
+SRC_URI[sha256sum] = "93be63a95f5ea7d5be2648709c777b07feb71d5fba651acb9cc05197465cb666"
+PROMO_TAG = "251120"
```
Take a look at other recipes and try reusing the name.

ricardosalveti commented on recipes-multimedia/camx/chicdk_1.0.1.bb:
```diff
@@ -0,0 +1,18 @@
+SRC_URI = "https://qartifactory-edge.qualcomm.com/artifactory/qsc_releases/software/chip/component/camx.qclinux.0.0/251120/prebuilt_yocto/${BPN}_${PV}_armv8-2a.tar.gz;subdir=${BPN}-${PV}"
+SRC_URI[sha256sum] = "ed134570cce02d17cee56b2ca81889f9ea78091451ed65b2e873053a9bd30bf3"
+
+DEPENDS += "camx camxcommon fastrpc camxlib glib-2.0 libxml2"
```
Add extra empty line.

ricardosalveti commented on recipes-multimedia/camx/chicdk_1.0.1.bb:
```diff
@@ -0,0 +1,18 @@
+SRC_URI[sha256sum] = "ed134570cce02d17cee56b2ca81889f9ea78091451ed65b2e873053a9bd30bf3"
+
+DEPENDS += "camx camxcommon fastrpc camxlib glib-2.0 libxml2"
+do_install() {
```
We can probably also add do_configure[noexec] = "1" and do_compile[noexec] = "1" as you are only interested on do_install.

ricardosalveti commented on recipes-multimedia/camx/chicdk_1.0.1.bb:
```diff
@@ -0,0 +1,18 @@
+    install -d ${D}${libdir}
+    cp -r ${S}/usr/lib/* ${D}${libdir}
+}
+FILES:${PN} = "/usr/lib/*"
```
It is probably not needed, /usr/lib is probably in the default FILES.

ricardosalveti commented on recipes-multimedia/camx/chicdk_1.0.1.bb:
```diff
@@ -0,0 +1,18 @@
+    cp -r ${S}/usr/lib/* ${D}${libdir}
+}
+FILES:${PN} = "/usr/lib/*"
+#Skips check for .so symlinks
```
Add whitespace after '#'

ricardosalveti commented on recipes-multimedia/camx/chicdk_1.0.1.bb:
```diff
@@ -0,0 +1,18 @@
+FILES:${PN} = "/usr/lib/*"
+#Skips check for .so symlinks
+INSANE_SKIP:${PN} = "already-stripped"
+#The modules require .so to be dynamicaly loaded
```
The modules? Which?

lumag commented on recipes-multimedia/camx/camx-kt_1.0.1.bb:
```diff
@@ -0,0 +1,25 @@
+}
+
+FILES_SOLIBSDEV = ""
+FILES:${PN} = "/usr/lib/*"
```
The  wildcard is to wide and includes files which are supposed to go to other packages. Please be more specific, describing only files which are required.

lumag commented on recipes-multimedia/camx/camxcommon_1.0.1.bb:
```diff
@@ -0,0 +1,19 @@
+SUMMARY = "Qualcomm camera common utility API used by camera driver"
```
This package includes files under `/usr/lib/qcs9100/`. Why are they a part of this package? Why is it split from camxlib?

lumag replied:
Also, we are definitely going to have several other versions of camx-something. Please name all packages according to platforms they support.

lumag reviewed (changes requested):
On top of everything. Please rename the packages in a sensible way.
Instead of `-kt` use `-kodiak`. Likewise add sensible suffix to `qcs9100`-related packages.

lumag commented on recipes-multimedia/camx/camxcommon_1.0.1.bb:
```diff
@@ -0,0 +1,30 @@
+SUMMARY = "Qualcomm camera common utility API used by camera driver"
+DESCRIPTION = "Collection of prebuilt libraries to support camera downstream functionality."
```
That can describe all other packages too

lumag commented on recipes-multimedia/camx/camxcommon_1.0.1.bb:
```diff
@@ -0,0 +1,30 @@
+}
+
+FILES_SOLIBSDEV += "${libdir}/qcs9100/*.so"
+FILES:${PN} += "${libdir}/qcs9100/*"
```
Okay, so, why did you extend `FILES_SOLIBSDEV`, but then `FILES:${PN}`?

gkhose-qipl replied:
> This is not an answer for the question that I have asked.

I'm sorry, my understanding of Yocto is limited, so I'm having difficulty grasping your question. Could you clarify what you mean?

lumag commented on recipes-multimedia/camx/camxcommon_1.0.1.bb:
```diff
@@ -0,0 +1,29 @@
+ARTIFACTORY_URL = "https://qartifactory-edge.qualcomm.com/artifactory/qsc_releases/software/chip/component/camx.qclinux.0.0/${PBT_BUILD_DATE}/prebuilt_yocto"
+
+SRC_URI = "${ARTIFACTORY_URL}/${BPN}_${PV}_armv8-2a.tar.gz"
+SRC_URI[sha256sum] = "6cf60e49e2a65ad6c65b48f05c63116b0b23652aabca29223f6881d0dab418cf"
```
What is going on? You've reuploaded the binary with the same name / PV and with a different checksum? This is a no-go!

```
SRC_URI = "https://qartifactory-edge.qualcomm.com/artifactory/qsc_releases/software/chip/component/camx.qclinux.0.0/251120/prebuilt_yocto/${BPN}_${PV}_armv8-2a.tar.gz;subdir=${BPN}-${PV}"
SRC_URI[sha256sum] = "71a381ae20e497f0d16789b22d10eeb1284558aebdf2cc6daa9855d05b807333"
```

lumag commented on recipes-multimedia/camx/camxcommon_1.0.1.bb:
```diff
@@ -0,0 +1,30 @@
+    cp -r ${S}/usr/lib/* ${D}${libdir}
+}
+
+FILES_SOLIBSDEV += "${libdir}/qcs9100/*.so"
```
Or, to be more explicit. I assume that apps and other libraries will have `libcamxsettingsmanager.so.0` and `libcamxcommonutils.so.0` in DT_NEEDED. However it's not so clear for the third library. `com.qti.settings.qcs9100.so` from its name looks more like a plugin. Which name is used to load it? Is it `com.qti.settings.qcs9100.so` or `com.qti.settings.qcs9100.so.0`?

lumag commented on recipes-multimedia/camx/camxcommon_1.0.1.bb:
```diff
@@ -0,0 +1,30 @@
+}
+
+FILES_SOLIBSDEV += "${libdir}/qcs9100/*.so"
+FILES:${PN} += "${libdir}/"
```
NAK

gkhose-qipl commented:
validation is not done, please hold on review,

gkhose-qipl commented on recipes-multimedia/camx/camxcommon_1.0.1.bb:
```diff
@@ -0,0 +1,30 @@
+SUMMARY = "Qualcomm camera common utility API used by all CamX components"
+DESCRIPTION = "Collection of prebuilt libraries to support common utility API used by all CamX components."
+LICENSE = "CLOSED"
```
just saw LICENSE PR is merged,
https://github.com/qualcomm-linux/meta-qcom/pull/1243
will make required changes

gkhose-qipl replied:
done

gkhose-qipl commented on recipes-multimedia/camx/camxcommon_1.0.1.bb:
```diff
@@ -0,0 +1,30 @@
+
+ARTIFACTORY_URL = "https://qartifactory-edge.qualcomm.com/artifactory/qsc_releases/software/chip/component/camx.qclinux.0.0/${PBT_BUILD_DATE}/prebuilt_yocto"
+
+SRC_URI = "${ARTIFACTORY_URL}/${BPN}_${PV}_armv8-2a.tar.gz"
```
done

lumag commented:
> Proposal : Could you please review proposal and update comments. For now looking for exception in the current form, As part of qcs615 enablement (end of Jan), this will be fixed as per proposal alignment

Also, why do you need more and more exceptions? Why can't the work be done properly from the beginning?

ricardosalveti commented:
> > > On top of everything. Please rename the packages in a sensible way. Instead of `-kt` use `-kodiak`. Likewise add sensible suffix to `qcs9100`-related packages.
> > 
> > 
> > Still being ignored
> 
> recipe name : qcm6490 target camxlib-v1_1.0.1.bb camx-v1_1.0.1.bb chicdk-v1_1.0.1.bb
> 
> lib path /usr/lib/camxv1/qcm6490
> 
> recipe name : qcs9100, qcs8300, qcs615, upcoming targets... camxcommon-v2_1.0.1.bb camxlib-v2_1.0.1.bb camx-v2_1.0.1.bb chicdk-v2_1.0.1.bb

v1 and v2 with the same versions? Makes no sense to me.

Is this supposed to be aligned for a certain family of socs? Can you describe what v1 and v2 actually means?

gkhose-qipl commented:
camxv1 and camxv2 are 2 different camera stack camxv1 support qcm6490, camxv2 support qcs9100, qcs8300 , qcs615 and new chipset on camxv2.

so we thought of having v1 and v2 in recipe name and as suggested by @lumag  we can install libs at ${libdir}/${SoC}

gkhose-qipl commented:
> ill those targets reuse `qcs9100` libs?

for qcs615 lib will be installed /usr/lib/qcs615

lumag commented:
/

> > ill those targets reuse `qcs9100` libs?
> 
> for qcs615 lib will be installed /usr/lib/qcs615

So, we logically have `camx-kodiak`, `camx-monaco` and `camx-talos`. Plus the overall `camx` frontend, selecting the correct implementation. Does that sound good?

gkhose-qipl commented:
> /
> 
> > > ill those targets reuse `qcs9100` libs?
> > 
> > 
> > for qcs615 lib will be installed /usr/lib/qcs615
> 
> So, we logically have `camx-kodiak`, `camx-monaco` and `camx-talos`. Plus the overall `camx` frontend, selecting the correct implementation. Does that sound good?

Yes, for every chipset libs are different and install at SOC specific paths

lumag commented:
> > /
> > > > ill those targets reuse `qcs9100` libs?
> > > 
> > > 
> > > for qcs615 lib will be installed /usr/lib/qcs615
> > 
> > 
> > So, we logically have `camx-kodiak`, `camx-monaco` and `camx-talos`. Plus the overall `camx` frontend, selecting the correct implementation. Does that sound good?
> 
> Yes, for every chipset libs are different and install at SOC specific paths

So, no need for v1/v2/vN and we are settled with the names.

gkhose-qipl commented on recipes-multimedia/camx/camxcommon_1.0.1.bb:
```diff
@@ -0,0 +1,31 @@
+LICENSE = "LICENSE.qcom-2"
+LIC_FILES_CHKSUM = "file://${UNPACKDIR}/usr/share/doc/${PN}/NO.LOGIN.BINARY.LICENSE.QTI.pdf;md5=7a5da794b857d786888bbf2b7b7529c8"
+
+SRC_URI = "https://qartifactory-edge.qualcomm.com/artifactory/qsc_releases/software/chip/component/camx.qclinux.0.0/${PBT_BUILD_DATE}/prebuilt_yocto/${BPN}_${PV}_armv8-2a.tar.gz"
```
> Okay... So, you did it again. Tarball changed, PV didn't. Could you please STOP doing that?

For the base PR, we often see that the artifactory and SHA are updated even when there are no changes in the camxcommon recipe. However, if other recipes are modified, the SHA for all recipes also changes. Therefore, we need to update the SHA for all recipes whenever any changes occur. We were considering updating the version with every change once the base PR is merged. Would that be acceptable?

lumag replied:
There was no **PR** in my sentence. No recipes. Stop reusing tarball names. Ever. It breaks OE caching. It breaks other possible ways to cache things. It breaks my mind, when I download yet another tarball pointed out by the recipe's URL and then face several files with the same basename. If I or somebody else downloads the tarball, wraps it in deb, rpm, apk or any other package, processes something then reports you an error, how would you determine, which one of the 1.0.1 tarballs had an issue? The rule of thumb is very, very simple: new upload means new tag, new version, new tarball name. If the contents doesn't change, find the ways to create binary reproducible tarballs (yes, it is possible, there are flags for that). But there should never be two tarballs with the same name and with different checksums.

lumag replied:
> As each distribution brings their own GCC, GLIBC and other differences, can we use same binaries for ubuntu distributions?

It's called ABI compatibility. In 99.9 % of the cases the binary compiled with the older GCC against the older Glibc will work with the newer ones. So, a typical approach would be to select some balanced old baseline and test that the binaries built with it still work with the newest one.

gkhose-qipl commented:
> * package provide common utility API used by all CamX components
> * Contains shared utilities and helper code used across multiple CamX modules.
> * Provides common functionality such as logging, basic data structures, and configuration handling.
> * Helps reduce duplication by centralizing generic logic that does not belong to any single feature.
> * Offers small, focused components that can be reused without pulling in heavy dependencies.

> Commit message, commit subject, package name, summary, description... Nothing changes, no improvememtns. Could you please fix those items for once?
> 
> For the commit subject. Does 'foo: add foo' sound good to you? I'd say, too many 'camxcommon' in one line.

We have tried to provide more descriptive commit messages. The discussion about the package name is still ongoing. I have added the summary and description, but could you clarify what exactly is expected for the summary and description?

lumag commented:
> > > > > > /
> > > > > > > > ill those targets reuse `qcs9100` libs?
> > > > > > > 
> > > > > > > 
> > > > > > > for qcs615 lib will be installed /usr/lib/qcs615
> > > > > > 
> > > > > > 
> > > > > > So, we logically have `camx-kodiak`, `camx-monaco` and `camx-talos`. Plus the overall `camx` frontend, selecting the correct implementation. Does that sound good?
> > > > > 
> > > > > 
> > > > > Yes, for every chipset libs are different and install at SOC specific paths
> > > > 
> > > > 
> > > > So, no need for v1/v2/vN and we are settled with the names.
> > > 
> > > 
> > > we need to understand what could be the possible way to install path and recipe name.
> > > 
> > > 1. camera v2 support multiple SOC, few are binary compatible (qcs9100 and qcs8100)
> > 
> > 
> > Provide virtual packages if required.
> > > 2. if there are different SOC how do we handle in binary compatible case (name of recipe, one recipe will install libs incase of binary compatible)
> > 
> > 
> > Compatibility of recipes or of the packages? I'd prefer having three separate recipes: kodiak, lemans+monaco and talos.
> > > 3. number of recipes increase exponentially  if we have soc specific recipes
> > 
> > 
> > Linarly, not exponentially. There is no exponential growth.
> 
> > > Compatibility of recipes or of the packages? I'd prefer having three separate recipes: kodiak, lemans+monaco and talos.
> 
> can we use camxv1, camxv2 , camxv3 for 3 different recipe names or can you help with some suggestion.

Of course not. What do v1, v2, v3 mean? Some virtual internal branches. So, no, it's not acceptable.

> 
> will check on virtual packages for (qcs9100 and qcs8300)

hint: `RPROVIDES`

lumag commented:
> > * package provide common utility API used by all CamX components
> > * Contains shared utilities and helper code used across multiple CamX modules.
> > * Provides common functionality such as logging, basic data structures, and configuration handling.
> > * Helps reduce duplication by centralizing generic logic that does not belong to any single feature.
> > * Offers small, focused components that can be reused without pulling in heavy dependencies.
> 
> > Commit message, commit subject, package name, summary, description... Nothing changes, no improvememtns. Could you please fix those items for once?
> > For the commit subject. Does 'foo: add foo' sound good to you? I'd say, too many 'camxcommon' in one line.
> 
> We have tried to provide more descriptive commit messages. The discussion about the package name is still ongoing. I have added the summary and description, but could you clarify what exactly is expected for the summary and description?

Commit subject: what actually happens. `camxcommon: add libs common to foo, bar and baz'. Or 'used by foo, bar and baz'. Commit message: English text. Not a bullet list. Not a list of changes in the commit, we can read patches. Describe why it needs to be done and why it needs to be done _this_ particular way.
Summary.  It should be a one-line description of the package. Let's see "Qualcomm camera common utility API used by all CamX components". What does it tell me? Used by which components? Is it only for the Lemans or will it be common between the platforms? Again, why is it not a part of the main camx package? What's the difference? But okay, it is more of the topic for the description. This text (i.e. several sentences) should provide a more detailed description of the package. How does one tell, which CamX packages to install? Which recipes to use? Here it would be nice to _explain_ what it is and why it is separated from the main camx lib package.

gkhose-qipl commented:
> > Of course not. What do v1, v2, v3 mean? Some virtual internal branches. So, no, it's not acceptable.
> 
> so what can be good way to name recipe. camx-kodiak, camx-talos , camx-lemans-monaco? if we have x,y,z target and all are binary compatible recipe name should be camx-x-y-z?

if we need to use RPROVIDES , what should be recipe name ?
camx-lemans-monaco_1.0.1.bb
RPROVIDES:${PN} += "camx-qcs8100 camx-qcs9100"

lumag commented:
As a side note I'll probably ask to use codenames for SoC subdirs. We have had enough troubles and confusion about qcs6490 vs qcs6490 vs sc7280, qrb5165 vs sm8250, etc. Using the codenames removes the topic as it points out that this CamX works for all kodiak instances, no matter if it is compute, mobile, IoT or something else.

lumag commented:
> > > > Colleagues. I'm starting to be really angry. Why did the version go **BACKWARDS**??
> > > 
> > > 
> > > done, updated to 1.0.3 as last version was 1.0.2
> > 
> > 
> > That's not an answer to the 'Why?' question.
> 
> @lumag , we initially kept the version as 1.0.0 to maintain a proper version history information. However, as per our discussion, updating the version to 1.0.3. In HWE, we have the camx version as 1.0.0, and there are currently no recipes hosted in meta-qcom. Now that we've renamed the recipes in meta-qcom, we're starting the version from 1.0.3,

Why are you again talking about recipes? The question is about release versions.

gkhose-qipl commented:
> > > > > Colleagues. I'm starting to be really angry. Why did the version go **BACKWARDS**??
> > > > 
> > > > 
> > > > done, updated to 1.0.3 as last version was 1.0.2
> > > 
> > > 
> > > That's not an answer to the 'Why?' question.
> > 
> > 
> > @lumag , we initially kept the version as 1.0.0 to maintain a proper version history information. However, as per our discussion, updating the version to 1.0.3. In HWE, we have the camx version as 1.0.0, and there are currently no recipes hosted in meta-qcom. Now that we've renamed the recipes in meta-qcom, we're starting the version from 1.0.3,
> 
> Why are you again talking about recipes? The question is about release versions.

release version for all QLI release was 1.0.0 in hwe.

anujm1 commented on recipes-multimedia/camx/camxcommon-lemans_1.0.3.bb:
```diff
@@ -0,0 +1,37 @@
+    install -d ${D}${datadir}/doc/${PN}
+
+    cp -r ${S}/usr/lib/* ${D}${libdir}
+    install -m 0644 ${S}/usr/share/doc/${PN}/NO.LOGIN.BINARY.LICENSE.QTI.pdf ${D}${datadir}/doc/${PN}
```
.. and here.

svankada commented on recipes-multimedia/camx/camxcommon-lemans_1.0.3.bb:
```diff
@@ -0,0 +1,37 @@
+
+do_install() {
+    install -d ${D}${libdir}
+    install -d ${D}${datadir}/doc/${PN}
```
Anuj, camera package name doesn't contain -native, -cross etc .. so PN and BPN does the same. We can make it as BPN if we ever add -native etc..

gkhose-qipl commented on recipes-multimedia/camx/camxcommon-lemans_1.0.3.bb:
```diff
@@ -0,0 +1,37 @@
+    This package provides the foundational utility libraries shared across all \
+    components of the Qualcomm Camera X (CamX). It is separated \
+    from the main qcom-camx package to resolve build dependencies, as these \
+    utilities are required by camx, chi-cdk, and camx-lib during compilation."
```
> Again, this doesn't make sense. We don't perform builds here.
> 
> It might help if you explain, what are the components. What is chi-cdk? Who uses it? What is the difference between camx and camx-lib? Doesn't camx depend on camx-lib? Does chi-cdk depend on camx? On camx-lib?

Dependency flow. chi-cdk->camx->camx-lib->camx-common 
camxcommon
camxlib (depends on camxcommon)
camx (depends on camxlib)
chicdk (depnds on camx)

chi-cdk:- (Camera Hardware Interface Customization Development Kit).

did not get "Who uses it? " can you please elaborate

lumag commented on dynamic-layers/openembedded-layer/recipes-multimedia/camx/camxlib-lemans_1.0.4.bb:
```diff
@@ -0,0 +1,129 @@
+
+# Disable configure and compile steps since this recipe uses prebuilt binaries.
+do_configure[noexec] = "1"
+do_compile[noexec] = "1"
```
You know, this makes me really sad. You've implemented this here, but didn't think about having the same bits in `camx-kodiak`. Why?

gkhose-qipl replied:
sorry it was missed in kodiak

lumag commented on dynamic-layers/openembedded-layer/recipes-multimedia/camx/camxlib-lemans_1.0.4.bb:
```diff
@@ -0,0 +1,129 @@
+    ${libdir}/camx/lemans/libiccprofile*${SOLIBSDEV} \
+    ${libdir}/camx/lemans/camera/components/com.qti.node*${SOLIBSDEV} \
+    ${libdir}/camx/lemans/camera/com.qti.sensor*${SOLIBSDEV} \
+    ${libdir}/camx/lemans/hw/com.qti.chi*${SOLIBSDEV} \
```
Who needs these files?

gkhose-qipl replied:
no one is going to use, it is just by product of compilation.
can you suggest way to ignore this without QA error ?

lumag commented on dynamic-layers/openembedded-layer/recipes-multimedia/camx/camxlib-lemans_1.0.4.bb:
```diff
@@ -0,0 +1,129 @@
+    install -d ${D}/etc/camera/test/NHX/
+	
+    cp -r ${S}/usr/lib/* ${D}${libdir}
+    cp -r ${S}/etc/camera/test/NHX/NHX.YUV_NV12_Prev_MaxRes.json ${D}/etc/camera/test/NHX/
```
`${sysconfdir}`

gkhose-qipl commented on dynamic-layers/openembedded-layer/recipes-multimedia/camx/camxlib-lemans_1.0.4.bb:
```diff
@@ -0,0 +1,129 @@
+    ${libdir}/camx/lemans/libhdr10plus*${SOLIBSDEV} \
+    ${libdir}/camx/lemans/libcamera_metadata*${SOLIBSDEV} \
+    ${libdir}/camx/lemans/libcamximageformatutils*${SOLIBSDEV} \ 
+    ${libdir}/camx/lemans/camera/components/com.qti.stats*${SOLIBSDEV} \
```
no one is going to use, it is just by product of compilation.
 FILES:${PN}-dev = " ${libdir}/camx/lemans/*${SOLIBSDEV} " is this  expected here?
 how else are we going to package .a and .so(unversioned) libs

lumag replied:
No, you need to `rm -f` unnecessary files.

nbobbaqcom commented on dynamic-layers/openembedded-layer/recipes-multimedia/camx/camxlib-lemans_1.0.4.bb:
```diff
@@ -0,0 +1,129 @@
+    ${libdir}/camx/lemans/libcamxhwnodecontext*${SOLIBSDEV} \
+    ${libdir}/camx/lemans/libcom.qti.eisv3*${SOLIBSDEV} \
+    ${libdir}/camx/lemans/libcamxsettingsmanager*${SOLIBSDEV} \
+    ${libdir}/camx/lemans/libopencv3a*${SOLIBSDEV} \
```
As OpenCV bins are distributed as part of CAMX stack the overall license LICENSE.QCOM-2 applies to all the components in CAMX stack. Since we are delivering OpenCV binaries we have the applicable NOTICES in NOTICE file.

gkhose-qipl replied:
3.1 version of openCV is being used & it is under BSD-3-Clause license

ricardosalveti commented:
In your commit message:

`camx:camx libraries for camera validation`

Remove the duplicated camx string.

```
The camx-lib package provides all camera libraries,
 shared across all components of the Qualcomm Camera X (CamX).
This recipe help to create separate camx packages.
3.1 version of openCV is being used.
```

Replace package with component or something else, package is confusing since the recipe will also generate packages which is later consumed by the image.

`This recipe help to create separate camx packages.`

Remove this, not really relevant.

ricardosalveti commented on dynamic-layers/openembedded-layer/recipes-multimedia/camx/camxlib-kodiak_1.0.4.bb:
```diff
@@ -0,0 +1,92 @@
+    install -m 0644 ${S}/usr/share/doc/${BPN}/NO.LOGIN.BINARY.LICENSE.QTI.pdf ${D}${datadir}/doc/chicdk-kodiak	
+}
+
+PACKAGE_BEFORE_PN += " camx-kodiak chicdk-kodiak"
```
No need to add the initial whitespace with +=.

ricardosalveti commented on dynamic-layers/openembedded-layer/recipes-multimedia/camx/camxlib-lemans_1.0.4.bb:
```diff
@@ -0,0 +1,91 @@
+SRC_URI[camx.sha256sum] = "3baea58276298602d5e9b980a814a089378f0c9095683503befa7c32eee01dba"
+SRC_URI[chicdk.sha256sum] = "bb73acd375abf114056c2fdfb345f648a3501a46ca292d5edf717a2ebae4b026"
+SRC_URI[camxcommon.sha256sum] = "9f028650c176e306da8f979adc52fd1010e588379fe53971bd909ef776e5519f"
+PBT_BUILD_DATE = "260102"
```
Add empty line here.

ricardosalveti commented on dynamic-layers/openembedded-layer/recipes-multimedia/camx/camxlib-lemans_1.0.4.bb:
```diff
@@ -0,0 +1,91 @@
+PBT_BUILD_DATE = "260102"
+S = "${UNPACKDIR}"
+
+DEPENDS += "glib-2.0 fastrpc protobuf libxml2"
```
Why do we have 'virtual/egl virtual/libgles2 virtual/libopencl1' for kodiak and not for lemans?

gkhose-qipl commented on dynamic-layers/openembedded-layer/recipes-multimedia/camx/camxlib-lemans_1.0.4.bb:
```diff
@@ -0,0 +1,129 @@
+    ${libdir}/camx/lemans/camera/com.qti.sensor*${SOLIBS} \
+    ${libdir}/camx/lemans/hw/com.qti.chi.*${SOLIBS} \
+    /etc/camera/test/NHX/ \
+    ${bindir}/ \
```
capturing images without sensor or tuning binaries is also not possible.Chi-cdk is also required to capture camera image.

lumag replied:
The apps don't depend on the CHICDK. It is an internal design of CamX. As such, we should only be pulling camx-kodiak & camx-lemans into the image. The rest must be handled by the packages themselves. Your libraries are doing tons of dlopens. Consider CamX being a normal, dynamically-linked set of libraries. Which package would depend on which?

gkhose-qipl commented on dynamic-layers/openembedded-layer/recipes-multimedia/camx/camxlib-kodiak_1.0.4.bb:
```diff
@@ -0,0 +1,92 @@
+
+    cp -r ${S}/usr/lib/* ${D}${libdir}
+    cp -r ${S}/usr/bin/* ${D}${bindir}
+
```
@lumag ,
correct, that is our aim.
for now removing unversioned libraries , will be added when compilation task arise.
